hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NamenodeFsck.java

```diff
@@ -1201,7 +1200,7 @@ private void lostFoundInit(DFSClient dfs) {
         lfInitedOk = true;
       }
     }  catch (Exception e) {
-      e.printStackTrace();
+      LOG.error(lfName + " dir init failed.", e);
       lfInitedOk = false;
     }
```

Comment:
why do we need to catch the exception itself, can't we just throw. If I catch the logic, we catch the exception and set ``lfInitedOk`` to `false`  and then in the calling method we throw
```
      if (!lfInitedOk) {
        throw new IOException("failed to initialize lost+found");
      }
```
Can't we directly throw. The client will get the exception, it will log or whatever rather than logging here

Reply:
Thank you for your guidance. It seems better to throw out exceptions so that the Client can also perceive specific exception information. I will immediately modify it